db/migrate/20220105024605_add_trigger_for_polymorphic_bookmark_columns_to_sync_data.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+class AddTriggerForPolymorphicBookmarkColumnsToSyncData < ActiveRecord::Migration[6.1]
+  def up
+    DB.exec <<~SQL
+      CREATE OR REPLACE FUNCTION sync_bookmarks_polymorphic_column_data()
+      RETURNS TRIGGER
+      LANGUAGE PLPGSQL AS $rcr$
+      BEGIN
+        IF NEW.for_topic
+        THEN
+          NEW.bookmarkable_id = (SELECT topic_id FROM posts WHERE posts.id = NEW.post_id);
+          NEW.bookmarkable_type = 'Topic';
+        ELSE
+          NEW.bookmarkable_id = NEW.post_id;
+          NEW.bookmarkable_type = 'Post';
+        END IF;
+        RETURN NEW;
+      END
+      $rcr$;
+    SQL
+
+    DB.exec <<~SQL
+      CREATE TRIGGER bookmarks_polymorphic_data_sync
+      BEFORE INSERT OR UPDATE OF post_id, for_topic ON bookmarks
+      FOR EACH ROW
+      EXECUTE FUNCTION sync_bookmarks_polymorphic_column_data();
+    SQL
+
+    # sync data that already exists in the table
+    DB.exec(<<~SQL)
```

Comment:
I am not sure if this needs to be in a post migrate or is fine to just be here?

Reply:
I think having it here is OK.